backend/controllers/vote.js

```diff
@@ -8,7 +8,7 @@ exports.vote = async (req, res) => {
   var { routeid } = req.params
   var { username, score } = req.body
 
-  if(!routeid || !username || !score || !Number.isInteger(score) || score < 0 || score > 5)
+  if(!routeid || !username || (!score && score!==0) || !Number.isInteger(score) || score < 0 || score > 5)
```

Comment:
If the score is `0`, the backend sees it as an invalid score because `!score` is evaluated as `!0` which is `true`. Needed to add an additional check here.